wildfly-ai/subsystem/src/main/java/org/wildfly/extension/ai/model/chat/ChatModelConnectionCheckerOperationHandler.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2024 JBoss by Red Hat.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.extension.ai.model.chat;
+
+import static org.jboss.dmr.ModelType.STRING;
+import static org.wildfly.extension.ai.AIAttributeDefinitions.USER_MESSAGE;
+
+import dev.langchain4j.model.chat.ChatLanguageModel;
+import org.jboss.as.controller.OperationContext;
+import org.jboss.as.controller.OperationFailedException;
+import org.jboss.as.controller.OperationStepHandler;
+import org.jboss.as.controller.SimpleOperationDefinitionBuilder;
+import org.jboss.as.controller.access.management.SensitiveTargetAccessConstraintDefinition;
+import org.jboss.as.controller.registry.ManagementResourceRegistration;
+import org.jboss.dmr.ModelNode;
+import org.wildfly.extension.ai.AILogger;
+import org.wildfly.service.capture.ValueExecutorRegistry;
+import org.wildfly.subsystem.resource.ResourceDescriptor;
+
+/**
+ *
+ * @author Emmanuel Hugonnet (c) 2024 Red Hat, Inc.
+ */
+public class ChatModelConnectionCheckerOperationHandler implements OperationStepHandler {
+    private static final String OPERATION_NAME = "chat";
+
+    public static void register(final ManagementResourceRegistration registration, final ResourceDescriptor descriptor, ValueExecutorRegistry<String, ChatLanguageModel> registry) {
+        registration.registerOperationHandler(SimpleOperationDefinitionBuilder.of(OPERATION_NAME, descriptor.getResourceDescriptionResolver())
+                .setParameters(USER_MESSAGE)
+                .setReplyType(STRING)
+                .setRuntimeOnly()
+                .setReadOnly()
+                .addAccessConstraint(SensitiveTargetAccessConstraintDefinition.READ_WHOLE_CONFIG)
+                .build(), new ChatModelConnectionCheckerOperationHandler(registry));
+    }
+
+    private final ValueExecutorRegistry<String, ChatLanguageModel> registry;
+
+    ChatModelConnectionCheckerOperationHandler(ValueExecutorRegistry<String, ChatLanguageModel> registry) {
+        this.registry = registry;
+    }
+
+    @Override
+    public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+        String userMessage = USER_MESSAGE.resolveModelAttribute(context, operation).asString();
+        ModelNode answer = registry.getExecutor(context.getCurrentAddressValue()).execute((ChatLanguageModel chatLanguageModel) -> {
```

Comment:
Just be aware that FunctionExecutor.execute(...) will return null if:
* the registry has no such entry
* the providing service is not yet started (or has since stopped).

Reply:
Ok, thanks